display/ssd1306.h

```diff
@@ -379,61 +395,61 @@ class StandardDisplayController : public DisplayControllerBase<Width, col_t>, Sa
         if (strchr(message_, '\n')) {
           display_->DrawText(message_, 0, 15, font);
         } else {
-	  // centered
+        // centered
           display_->DrawText(message_, 0, HEIGHT / 2 + 7, font);
         }
         next_screen_ = SCREEN_DEFAULT;
-	// STDERR << "MESSAGE, millis = " << font_config.ProffieOSFontImageDuration << "\n";
-        return font_config.ProffieOSFontImageDuration;
+        // STDERR << "MESSAGE, millis = " << font_config.ProffieOSFontImageDuration << "\n";
+        return 3000; 
```

Comment:
In repeated testing, I personally found this to be a good number. Not too short not too long.
It's arbitrary. Whatever you think is best.

Reply:
The problem is that it's neither backwards compatible, nor configurable.
If someone was using ProffieOSFontImageDuration for this and relied on it, it won't work anymore and they can't change it. (Without changing the code.)

Not saying that 3000 is a bad choice really,  I just want to think through the alternatives, like maybe another configuration variable?

Reply:
(and whether backwards compatibility would be good or bad.)

Reply:
I'm trying to remember why I made this change. I think it had to do with the Error messages. Since they use this to display, but there's no wav for them, if the fontconfig duration was set to zero for auto sync, it wouldn't show the message.
Can we check if the incoming message is from ProffieOSErrors then use ProffieOSFontImageDuration, else 3000?

Reply:
That seems more like a problem with the autosync feature.
Like maybe we should change autosync so that if you specify a negative number, then we use the wav length, and if there is no wav file, we negate the number and use that?

Reply:
Of course that could work, I feel it's a little awkward setting -3000, considering the standard for other things to sync (IgnitionTime<>) is zero. Maybe -1, but still.

> like maybe another configuration variable?

This probably makes the most sense; to break text messages out of the Image Duration anyway as they're 2 different things anyway really, right?
We could just add something like font_config.ProffieOSTextMessageDuration(?) that is a time for all OLED printed messages.
And how about meet in the middle and make it's default 3500ms?
Just tested that and it works great and seems plausible to me.

Reply:
Seems fairly reasonable.
And if that's not set we can fall back to ProffieOSFontImageDuration for backwards compatibility.

Reply:
I don't know the syntax to check if it's set in font_config vs in hybrid_font.

Reply:
The easiest way is to make the default 0 in hybrid_font, then you can use if statements to see if it's zero or not.

Reply:
Hmm

> if (font_config.ProffieOSTextMessageDuration) is not set we can fall back to ProffieOSFontImageDuration for backwards compatibility.

Hmm, this puts us back to the problem of if FontImageDuration is set to Zero for syncing to wav.

Unless we do an else-if and then double fall back to a static value like 3500?
So:
Set TextMessageDuration default to 0.
Then if font_config.ProffieOSTextMessageDuration !=0, use it
else if font_config.ProffieOSFontImageDuration is not Zero, use it,
else 3500.
Does that work?

Reply:
How about we just make the default -1?
That way we can tell the difference between zero and unset.

Reply:
Meaning so that zero is a useable option to (not) display text?
ok. But I don't see the point.

Reply:
This is done, i left off testing and didn't get back here yet.
If ProffieOSFontImageDuration has been set to 0 for sync purposes, then it would need a fallback default time.
This covers the below comment for STARTUP_SCREEN as well.
The -1 feels clunky though.